clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -871,10 +896,11 @@ invariant_checker_t::check_for_pc_discontinuity(
             TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
                     shard->file_type_) ||
             // Regular fall-through.
-            (prev_instr_trace_pc + shard->prev_instr_.instr.size == memref.instr.addr) ||
+            (prev_instr.instr.addr + prev_instr.instr.size == current_memref_addr) ||
             // String loop.
-            (prev_instr_trace_pc == memref.instr.addr &&
-             (memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH ||
+            (prev_instr.instr.addr == current_memref_addr &&
+             (memref_is_kernel_event_marker ||
```

Comment:
As discussed offline, we should not try to add to this "string loop" condition. There are other parts to this condition that apply only to the string loop case